### ISSUE TITLE
[BUG] Playwright clashes with jacoco (Java Code Coverage)

### ISSUE THREAD

**Issue by jhyot** [microsoft/playwright-java#802]
**Context:**
- Playwright Version: 1.18
- Operating System: Linux
- Browser: Chromium

**Describe the bug**

The newest version seems to clash with jacoco which is a popular java code coverage tool. Jacoco (as all code coverage tools usually do) instruments the java bytecode with code coverage tracking instructions. Playwright internals does not seem to like that anymore. See the stacktrace.

We upgraded Playwright from 1.14.1, in which everything worked, to 1.18.0, which now fails. I don't know in which exact version the problem was introduced.

Note that using the code coverage tracker in our specific case is probably a mistake (we don't need code coverage for our playwright tests), but because jacoco is extremely popular, any library should make sure that it works correctly with it.

**Stacktrace**
```
com.microsoft.playwright.PlaywrightException: Internal error
	at app//com.microsoft.playwright.impl.Utils.convertType(Utils.java:69)
	at app//com.microsoft.playwright.impl.LocatorImpl.innerText(LocatorImpl.java:210)
	at app//com.microsoft.playwright.Locator.innerText(Locator.java:2079)
	at <... redacted client code ...>
	at java.base@11.0.13/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base@11.0.13/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base@11.0.13/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base@11.0.13/java.lang.reflect.Method.invoke(Method.java:566)
	at app//org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at app//org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at app//org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at app//org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at app//org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at app//org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at app//org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at app//org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at app//org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at app//org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at app//org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at app//org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at app//org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at app//org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at app//org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at app//org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at app//org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at app//org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at app//org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at app//org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at app//org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at app//org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:110)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:38)
	at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:62)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
	at java.base@11.0.13/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base@11.0.13/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base@11.0.13/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base@11.0.13/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy5.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker$2.run(TestWorker.java:176)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:129)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:100)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:60)
	at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:133)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:71)
	at app//worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
	at app//worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)
Caused by: java.lang.RuntimeException: Unexpected field modifiers: com.microsoft.playwright.Frame.InnerTextOptions.$jacocoData, modifiers: 4234
	at com.microsoft.playwright.impl.Utils.convertType(Utils.java:55)
	... 54 more
```

**Comment by jhyot** [comment]
Two additions:

* I can confirm that the bug does **not** occur in 1.17.2, so it is a regression in the latest minor version.
* I was looking how to conditionally disable jacoco for a specific gradle task, but it seems to be so easy. Jacoco is a built-in feature in gradle an in a way "always-on". This makes it even more important that playwright fixes the issue. For now we are rolling back to 1.17.2, until the problem is fixed.

**Comment by yury-s** [comment]
Would you be able to share a project where this is reproducible?

```
Caused by: java.lang.RuntimeException: Unexpected field modifiers: com.microsoft.playwright.Frame.InnerTextOptions.$jacocoData, modifiers: 4234
```
This error means that jacoco tampers with Playwright data structures and breaks its invariants. Sounds like a problem with their library as they are changing original modifiers / class structure but it might be easy to fix on our end too, we just need a repro.

**Comment by jhyot** [comment]
[playwright-jacoco.zip](https://github.com/microsoft/playwright-java/files/8022873/playwright-jacoco.zip)

Repro:
* Make sure you have a JDK installed and on the path (I tested with JDK 11)
* Unpack the attached zip file into a new directory
* Navigate into the unpacked folder and run `./gradlew test --stacktrace`

I don't know much about jacoco. In case you think they are doing something wrong/weird, it might be worth starting a discussion on their issue tracker: https://github.com/jacoco/jacoco

**Comment by jhyot** [comment]
There is a FAQ from jacoco about this problem:
https://www.eclemma.org/jacoco/trunk/doc/faq.html

> My code uses reflection. Why does it fail when I execute it with JaCoCo?
>
> To collect execution data JaCoCo instruments the classes under test which adds two members to the classes: A private static field `$jacocoData` and a private static method `$jacocoInit()`. Both members are marked as synthetic.
Please change your code to ignore synthetic members. This is a good practice anyways as also the Java compiler creates synthetic members in certain situation.

The synthetic fields in question can be checked with `isSynthetic()`.

**Comment by yury-s** [comment]
@jhyot the fix is in, could you verify it with 1.19.0-SNAPSHOT version?

**Comment by jhyot** [comment]
@yury-s Tested with 1.19.0 release version, it is working.
